units/src/amount/signed.rs

```diff
@@ -55,20 +55,65 @@ impl SignedAmount {
     pub const ONE_SAT: Self = SignedAmount(1);
     /// Exactly one bitcoin.
     pub const ONE_BTC: Self = SignedAmount(100_000_000);
-    /// The maximum value allowed as an amount. Useful for sanity checking.
+    /// The maximum value allowed as an amount.
     pub const MAX_MONEY: Self = SignedAmount(21_000_000 * 100_000_000);
     /// The minimum value of an amount.
     pub const MIN: Self = SignedAmount(-21_000_000 * 100_000_000);
     /// The maximum value of an amount.
     pub const MAX: Self = SignedAmount::MAX_MONEY;
 
     /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
-    pub const fn from_sat(satoshi: i64) -> SignedAmount { SignedAmount(satoshi) }
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bitcoin_units::SignedAmount;
+    /// # let sat = -100_000;
+    /// let amount = SignedAmount::from_sat(sat).expect("sat is less than MAX_MONEY");
+    /// assert_eq!(amount.to_sat(), sat);
+    /// ```
+    pub const fn from_sat(satoshi: i64) -> Result<SignedAmount, OutOfRangeError> {
+        if satoshi < Self::MIN.to_sat() {
+            Err(OutOfRangeError { is_signed: true, is_greater_than_max: false })
+        } else if satoshi > Self::MAX_MONEY.to_sat() {
+            Err(OutOfRangeError { is_signed: true, is_greater_than_max: true })
+        } else {
+            Ok(SignedAmount(satoshi))
+        }
+    }
+
+    /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
+    ///
+    /// Caller to guarantee that `satoshi` is within valid range.
+    ///
+    /// See [`Self::MIN`] and [`Self::MAX_MONEY`].
```

Comment:
Maybe my version was overly verbose, but I think it makes sense to emphasize that if using this function, there program could misbehave in unpredictable ways.

Reply:
s/there/their

Reply:
Thanks, I disagree.